### PR TITLE
SR-79 Break out models.py and crud.py to new folders

### PR DESCRIPTION
**Pull request by jakethrasher**
# Task

I updated the file structure by migrating the code in `models.py` and `crud.py` to `models` and `services`. 

Most of the changes involve updating imports. 

I added comments to `models/__init__.py` and `services/__init__.py` for suggestions on importing from the new packages.

We can discuss these changes as a team and make sure everyone signs off on them.

All unit tests are passing.

[SR-79](https://incredihire-academy.atlassian.net/browse/SR-79)

[Backend patterns doc](https://docs.google.com/document/d/1EWSyKXvgutzK7QA8PH6iQgnl0_7q4QQpmqjSTBY24Vg/edit)

# Author Tasks

## Acceptance Criteria
Please include the acceptance criteria from the ticket here or select N/A.
- [x] N/A

### CODE REVIEW

File: backend/app/models/item.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import uuid
+from typing import TYPE_CHECKING, Optional
+
+from sqlmodel import Field, Relationship, SQLModel
+
+if TYPE_CHECKING:
```
Comment (mcverter) [inline backend/app/models/item.py:6]: I didn't know about this workaround.  Very impressive!

File: backend/app/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    User,
+    UserCreate,
+    UserPublic,
+    UserRegister,
+    UsersPublic,
+    UserUpdate,
+    UserUpdateMe,
+)
+
+# https://realpython.com/python-all-attribute/#names-from-a-package
```
Comment (mcverter) [inline backend/app/models/__init__.py:28]: this is great. thanks

File: backend/app/models/auth.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    access_token: str
+    token_type: str = "bearer"
+
+
+# Contents of JWT token
+class TokenPayload(SQLModel):
+    sub: str | None = None
+
+
+class NewPassword(SQLModel):
```
Comment (Eugene-Polonsky) [inline backend/app/models/auth.py:20]: I wonder if we should prioritize ripping all this functionality out. We're carrying this code from one place to another, bearing the cost, while we know it's all obsolete.

File: backend/app/services/login.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import app.services.users as users_service
+from app.core.security import verify_password
+from app.models import User
+
+
+def authenticate(*, session: Session, email: str, password: str) -> User | None:
+    db_user = users_service.get_user_by_email(session=session, email=email)
+    if not db_user:
+        return None
+    if not verify_password(password, db_user.hashed_password):
```
Comment (Eugene-Polonsky) [inline backend/app/services/login.py:12]: this seems obsolete.

File: backend/app/models/auth.py (hunk 3 above)
Comment (jakethrasher) [reply backend/app/models/auth.py:20]: That is a great point. Do you think I should do the work an include it in this PR?

**Pull request merged**